rasa/shared/utils/io.py

```diff
@@ -269,7 +271,10 @@ def read_yaml_file(filename: Union[Text, Path]) -> Union[List[Any], Dict[Text, A
     Args:
         filename: The path to the file which should be read.
```

Comment:
optional (as it's not your change): Document the exception in the docstring. (ideally also the `Returns:`)